openapi3gen/openapi3gen_test.go

```diff
@@ -590,3 +590,43 @@ func ExampleNewSchemaRefForValue_recursive() {
 	//   "type": "object"
 	// }
 }
+
+type ID [16]byte
+
+// T implements SetSchemar, allowing it to set an OpenAPI schema.
+type T struct {
+	ID ID `json:"id"`
+}
+
+func (_ *ID) SetSchema(schema *openapi3.Schema) {
+	schema.Type = &openapi3.Types{"string"} // Assuming this matches your custom implementation
+	schema.Format = "uuid"
+}
+
+func ExampleID_SetSchema() {
```

Comment:
Almost! See here you define an `Example` for the `SetSchema` method of the `ID` type. It doesn't help because these type and method live in test files so don't end up in the generated docs. Instead this should be named `ExampleSetSchemar`.

I'll amend this myself, no worries.